engine/src/main/java/org/pentaho/di/job/entries/deletefiles/JobEntryDeleteFiles.java

```diff
@@ -302,10 +303,13 @@ boolean processFile( String path, String wildcard, Job parentJob ) {
     boolean isDeleted = false;
     FileObject fileFolder = null;
 
+
+
     try {
       fileFolder = KettleVFS.getFileObject( path, this );
 
-      if ( fileFolder.exists() ) {
+      if ( fileFolder.exists() && new TextFileSelector( fileFolder.toString(), wildcard, parentJob ).hasAccess(fileFolder)) {
```

Comment:
hasAccess only checks if the file is readable, but we want to be able to delete it.  Have you verified that this works if the file is readable to the current user but owned by another user, and the current user does not have permission to delete?
You might need to use SecurityManager.checkDelete or something similar instead.

A unit or integration test for this would be a good idea.

Reply:
Resolved as discussed